Sources/NIOCore/ByteBuffer-aux.swift

```diff
@@ -143,6 +167,31 @@ extension ByteBuffer {
             return String(decoding: UnsafeRawBufferPointer(fastRebase: pointer[range]), as: Unicode.UTF8.self)
         }
     }
+    
+    /// Get the string at `index` from this `ByteBuffer` decoding using the UTF-8 encoding. Does not move the reader index.
+    /// The selected bytes must be readable or else `nil` will be returned.
+    ///
+    /// - parameters:
+    ///     - index: The starting index into `ByteBuffer` containing the null terminated string of interest.
+    /// - returns: A `String` value deserialized from this `ByteBuffer` or `nil` if there isn't a complete null-terminated string,
+    ///            including null-terminator, in the readable bytes after `index` in the buffer
+    public func getNullTerminatedString(at index: Int) -> String? {
+        guard let stringLength = self.getNullTerminatedStringLength(at: index) else {
+            return nil
+        }
+        return self.getString(at: index, length: stringLength)
+    }
+    
+    private func getNullTerminatedStringLength(at index: Int) -> Int? {
+        guard self.readerIndex <= index, index < self.readerIndex + self.readableBytes else {
+            return nil
+        }
+
+        guard let endIndex = self.readableBytesView[index...].firstIndex(where: { $0 == 0 }) else {
```

Comment:
Mild nit but we don't need to stress the optimiser with the closure: we can just do `firstIndex(of: 0)`.